### DISCUSSION TITLE
How to call c macro with parameters

### DISCUSSION THREAD

**Discussion by Blackrose** [vlang/v#18307] [Questions and Answers] [answered]
Hi all, I try to use v to read/write data through spidev0.0 device. In c, the write data will called ioctl function.Below is example
```c
ioctl(fd, SPI_IOC_MESSAGE(1), &tr);
```
the `SPI_IOC_MESSAGE` is a macro defined in file /usr/include/linux/spi/spidev.h. When I try to use v to call this macro, the v shows not found the function. I also try to call macro without parameter, it's good.  
```v
 error: unknown function: C.SPI_IOC_MESSAGE
rc := int(C.ioctl(this.fd, C.SPI_IOC_MESSAGE(1), &transfer))
```
Does the v not support call c macro with parameter? If so, how can this situation be resolved?

**Comment by JalonSolov** [comment, marked as answer]
You have to define `C.SPI_IOC_MESSAGE` in your V code, for V to recognize it.  V does not parse `.h` files, so it cannot know about it otherwise.
